docs/proposals/integrations_release_cycle.md

```diff
@@ -0,0 +1,90 @@
+# Integrations release cycle
+
+- Authors: Massimiliano Pippi
+- Date: 2018-12-12
+- Status: draft
+- [Discussion](https://github.com/DataDog/integrations-core/pull/2741)
+
+## Overview
+
+Each agent based integration has its own version number and can be released as
+a Python package at any given time. This document outlines what the release
+cycle will be for each integration.
+
+## Problem
+
+At this moment the release cycle for most of the integration is not formalized
+and there's no process in place to keep it consistent and reliable. Once a feature
+or a bugfix is merged on `master`, different things may happen:
+
+- A release is immediately triggered.
+- A release is triggered after merging other, maybe unrelated changes for the same integration.
+- A release is triggered at a later time without any specific deadline.
+
+This causes a number of issues for all actors involved:
+
+- Engineers have to improvise on a case by case basis.
+- Contributors have no idea if changes within a PR will be released and when.
+- Users don't know when a feature or a bugfix will be actually available.
+
+## Constraints
+
+1. Each integration must keep its own release cycle.
+2. The solutions must work for integrations from `integrations-extra` as well.
+
+### A note about testing
+
+At this moment, this is the list of automated checks performed for any PR and
+when something is merged in the `master` branch:
+
+- ensure the list of checks shipped with the agent is correct
+- validate config files are well formed (YAML, presence of an instance, etc)
+- validate the dependencies declared in the check are compatible with what the Agent ships
+- validate manifest file is valid (both JSON format and contents)
+- validate metadata (same as manifest)
+- validate service-checks (same as manifest)
+- unit and integrations test + test coverage (the Datadog Agent is not involved)
+- benchmarks (at the moment they don't fail but we could stop a release if a slowdown is found)
+
+This is **not** enough to ensure an integration can run on any supported platform
```

Comment:
Would you mind to give more precise examples here?

Reply:
it should be better now